### ISSUE TITLE
TFS service hook integration

### ISSUE THREAD

**Issue by Kemyke** [mattermost/platform#1880]
Since this fix https://github.com/mattermost/platform/pull/1817 we can successfully integrate with TFS 2015. But we have a problem with the formatting of the message. 
![mattermosttfsformatting](https://cloud.githubusercontent.com/assets/5611190/12293593/ea183802-b9f4-11e5-982d-58311602661a.png)

As you can see on the screenshot the link diplayed wrong. 
The TFS slack integration message looks like this:

```
Method: POST
URI: https://******/hooks/***********
HTTP Version: 1.1
Headers:
{
  Content-Type: application/json; charset=utf-8
}
Content:
{
  "attachments": [
    {
      "fields": [
        {
          "title": "Area",
          "value": "FabrikamCloud",
          "short": true
        },
        {
          "title": "Iteration",
          "value": "FabrikamCloud\\Release 1\\Sprint 1",
          "short": true
        },
        {
          "title": "State",
          "value": "New",
          "short": true
        },
        {
          "title": "Reason",
          "value": "New defect reported",
          "short": true
        },
        {
          "title": "Severity",
          "value": "3 - Medium",
          "short": true
        }
      ],
      "pretext": "<http://fabrikam-fiber-inc.visualstudio.com/web/wi.aspx?pcguid=74e918bf-3376-436d-bd20-8e8c1287f465&id=5|Bug #5> (Some great new idea!) created by Jamal Hartnett.",
      "mrkdwn_in": [
        "pretext"
      ],
      "fallback": "Bug #5 (Some great new idea!) created by Jamal Hartnett.\r\n(http://fabrikam-fiber-inc.visualstudio.com/web/wi.aspx?pcguid=74e918bf-3376-436d-bd20-8e8c1287f465&id=5)\r\n\r\n- Area: FabrikamCloud\r\n- Iteration: FabrikamCloud\\Release 1\\Sprint 1\r\n- State: New\r\n- Assigned to: \r\n- Comment: \r\n- Severity: 3 - Medium\r\n"
    }
  ]
}
```

It is really hard to read these webhook messages in this format.

**Comment by it33** [comment]
Thanks @Kemyke, @apskim, just curious, is this duplicate of https://github.com/mattermost/platform/issues/1868 or am I missing something?

**Comment by apskim** [comment]
@it33 yep, it's the same issue.

**Comment by it33** [comment]
Thanks @apskim, closing as duplicate, did you want to take a shot at addressing it in your PR? 

We can't merge until next week because of the localization work that's going on right now...

**Comment by apskim** [comment]
@it33 My PR (#1867) already fixes that. It's OK to wait, this is not a critically urgent issue :)

**Comment by it33** [comment]
Ah cool, sorry, thanks! :)

**Comment by Kemyke** [comment]
@ScottHutchinson as mattermost support the same API as Slack you can easily do it with VSTS/Azure DevOps or on-premises TFS. Check this doc: https://docs.microsoft.com/en-us/azure/devops/service-hooks/services/slack?view=azure-devops

**Comment by ScottHutchinson** [comment]
@Kemyke Thanks. I've been trying that, but so far there is some mismatch with http vs. https that I am unable to resolve. I have both mattermost server and TFS using http, but I still get an error similar to this: https://github.com/mattermost/mattermost-server/issues/3869

**Comment by esiemiat** [comment]
I have an on premise Azure DevOps instance and am trying to use the Slack service hook to communicate to MatterMost. I was curious if there is a way to specify the channel the messages should post to or am I limited to whatever channel the Incoming WebHook is configured with on the MatterMost side of things?
